src/Components/Server/src/Microsoft.AspNetCore.Components.Server.csproj

```diff
@@ -91,11 +91,27 @@
   </ItemGroup>
 
   <PropertyGroup>
-    <BlazorServerJSFile>..\..\Web.JS\dist\$(Configuration)\blazor.server.js</BlazorServerJSFile>
+    <!-- Microsoft.AspNetCore.Components.Web.JS.npmproj always capitalizes the directory name. -->
+    <BlazorServerJSFile
+        Condition=" '$(Configuration)' == 'Debug' ">..\..\Web.JS\dist\Debug\blazor.server.js</BlazorServerJSFile>
+    <BlazorServerJSFile
+        Condition=" '$(Configuration)' != 'Debug' ">..\..\Web.JS\dist\Release\blazor.server.js</BlazorServerJSFile>
   </PropertyGroup>
 
-  <ItemGroup>
-    <EmbeddedResource Include="$(BlazorServerJSFile)" LogicalName="_framework/%(Filename)%(Extension)" />
-    <EmbeddedResource Include="$(BlazorServerJSFile).map" LogicalName="_framework/%(Filename)%(Extension)" Condition="Exists('$(BlazorServerJSFile).map')" />
-  </ItemGroup>
+  <!-- blazor.server.js should exist after Microsoft.AspNetCore.Components.Web.JS.npmproj builds. Fall back if not. -->
+  <Target Name="_CheckBlazorServerJSPath" AfterTargets="ResolveProjectReferences"
+      Condition=" !EXISTS('$(BlazorServerJSFile)') ">
+    <Warning Text="'$(BlazorServerJSFile)' does not exist. Falling back to checked-in copy." />
+    <PropertyGroup>
+      <BlazorServerJSFile>..\..\Web.JS\dist\Release\blazor.server.js</BlazorServerJSFile>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="_AddEmbeddedBlazor" AfterTargets="_CheckBlazorServerJSPath">
+    <ItemGroup>
+      <EmbeddedResource Include="$(BlazorServerJSFile)" LogicalName="_framework/%(Filename)%(Extension)" />
+      <EmbeddedResource Include="$(BlazorServerJSFile).map" LogicalName="_framework/%(Filename)%(Extension)"
+          Condition="Exists('$(BlazorServerJSFile).map')" />
+    </ItemGroup>
```

Comment:
The problem leading to the test failures is in this target. I've downloaded a binary log to find the root cause.

When building in `main`, Microsoft.AspNetCore.Components.Server contains the following resources
![image](https://user-images.githubusercontent.com/6431421/107586446-c34fbc00-6bb4-11eb-9a6d-ab9733acbd91.png)
and the embedded manifest contains
``` xml
<?xml version="1.0" encoding="utf-8" standalone="yes"?
<Manifest>
  <ManifestVersion>1.0</ManifestVersion>
  <FileSystem>
...
    <Directory Name="_framework">
      <File Name="blazor.server.js">
        <ResourcePath>_framework/blazor.server.js</ResourcePath>
      </File>
    </Directory>
  </FileSystem>
</Manifest>
```

But, the version created in this PR skips the blazor.server.js filename both in the resources
![image](https://user-images.githubusercontent.com/6431421/107586748-67d1fe00-6bb5-11eb-988c-e1c053528abf.png)
and the manifest
``` xml
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Manifest>
  <ManifestVersion>1.0</ManifestVersion>
  <FileSystem>
...
    <Directory Name="_framework">
      <File Name="">
        <ResourcePath>_framework/</ResourcePath>
      </File>
    </Directory>
  </FileSystem>
</Manifest>
```